src/main/java/ch/njol/skript/expressions/ExprMaxStack.java

```diff
@@ -1,62 +1,116 @@
-/**
- *   This file is part of Skript.
- *
- *  Skript is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Skript is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Copyright Peter Güttinger, SkriptLang team and contributors
- */
 package ch.njol.skript.expressions;
 
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
 import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
-import org.bukkit.Material;
-import org.bukkit.inventory.ItemStack;
+import ch.njol.util.Math2;
+import ch.njol.util.coll.CollectionUtils;
 
-/**
- * @author joeuguce99
- */
 @Name("Maximum Stack Size")
-@Description("The maximum stack size of the specified material, e.g. 64 for torches, 16 for buckets, and 1 for swords.")
-@Examples("send \"You can only pick up %max stack size of player's tool% of %type of (player's tool)%\" to player")
-@Since("2.1")
-public class ExprMaxStack extends SimplePropertyExpression<ItemType, Long> {
+@Description({
+	"The maximum stack size of an item (e.g. 64 for torches, 16 for buckets, 1 for swords, etc.) or inventory.",
+	"In 1.20.5+, the maximum stack size of items can be changed to any integer from 1 to 99, and stacked up to the maximum stack size of the inventory they're in."
+})
+@Examples({
+	"send \"You can hold %max stack size of player's tool% of %type of player's tool% in a slot.\" to player",
+	"set the maximum stack size of inventory of all players to 16",
+	"add 8 to the maximum stack size of player's tool",
+	"reset the maximum stack size of {_gui}"
+})
+@Since("2.1, INSERT VERSION (changeable, inventories)")
+@RequiredPlugins("Spigot 1.20.5+ (changeable)")
```

Comment:
```suggestion
@RequiredPlugins("Minecraft 1.20.5+ (changeable)")
```

Reply:
Both term `Minecraft` and `Spigot` are acceptable based on past debates between members — unless a member objects, I’m marking this as resolved